spec.md

```diff
@@ -193,6 +194,7 @@ To pull a blob, perform a `GET` request to a URL in the following form:
 A GET request to an existing blob URL MUST provide the expected blob, with a response code that MUST be `200 OK`.
 A successful response SHOULD contain the digest of the uploaded blob in the header `Docker-Content-Digest`.
 If present, the value of this header MUST be a digest matching that of the response body.
+Clients SHOULD verify that the response body matches the requested digest and the response header digest.
```

Comment:
The registry could return a header digest with another algorithm, in that case the client should not trust the digest if it does not match but would need to verify the manifest against the client's digest reference and registries digest (if the client intends to store the content by the registry provided digest).

Reply:
What I meant to say with this sentence is that 

1. clients should _always_ verify the requested digest
2. if a response header digest is present, clients should verify that, too
3. implicitly, a blob should not be stored at all if it fails any of the above

Do you think I should make (3) explicit?

Reply:
From our discussions in yesterday's meeting, we didn't feel it necessary for clients to verify a potentially different digest algorithm, particularly if the server supports different algorithms than that client supports, and the client didn't request them.

```suggestion
Clients SHOULD verify that the response body matches the requested digest.
```

Reply:
I agree that clients should not need to verify the digest header if they're not using it, but should we ask them to do it when they use it, similar to the requirements for the manifest?

> Most clients MAY ignore the value, but if it is used, the client MUST verify the value [...]

Reply:
I don't see the value-add in verifying the server provided header. This digest is always known by the client, and needs to match the descriptor value in a manifest. If the server gives a different algorithm, there's no workflow where the pulling client should use that digest even if it verified it.

Reply:
I'd still like to change it. But we can add a line similar to 182 to cover any clients that later find a use case for the data.

```suggestion
Most clients MAY ignore the value, but if it is used, the client MUST verify the value matches the returned response body.
Clients SHOULD verify that the response body matches the requested digest.
```